uploadNeuroDB/NeuroDB/MRIProcessingUtility.pm

```diff
@@ -1770,72 +1804,96 @@ or a phantom
 
 sub validateCandidate {
     my $this = shift;
-    my ($subjectIDsref)= @_;
-    my $CandMismatchError = undef;
+    my ($subjectIDsref, $upload_id) = @_;
+
+    my ($CandMismatchError, $message);
+    my $pscID       = $subjectIDsref->{'PSCID'};
+    my $candID      = $subjectIDsref->{'CandID'};
+    my $visit_label = $subjectIDsref->{'visitLabel'};
+    my $dbh         = ${$this->{'dbhr'}};
+
+
+    ############################################################
+    ################ No Checking if the subject is Phantom #####
+    ############################################################
+    if ($subjectIDsref->{'isPhantom'}) {
+        # CandID/PSCID errors don't apply to phantoms, so we don't
+        # want to trigger the check which aborts the insertion
+        $CandMismatchError = undef;
+        return $CandMismatchError;
+    }
+
     
     #################################################################
     ## Check if CandID exists
     #################################################################
     my $query = "SELECT CandID, PSCID FROM candidate WHERE CandID=?";
-    my $sth = ${$this->{'dbhr'}}->prepare($query);
-    $sth->execute($subjectIDsref->{'CandID'});
-    print "candidate id " . $subjectIDsref->{'CandID'} . "\n" 
-	if ($this->{verbose});
+    my $sth   = $dbh->prepare($query);
```

Comment:
I think the entire check block could be simplified. I would only query the DB once, like this:

```sql
SELECT c1.CandID, c2.PSCID
FROM candidate c1 
LEFT JOIN candidate c2 ON (c1.CandID=c2.CandID AND c2.PSCID=?)
WHERE c1.CandID=?
```

If you get no results, then the CandID is invalid. If you get a result but the PSCID column is NULL, then the CandID and PSCID do not match. There is no need to check whether the provided PSCID exists: validating that it matches is enough IMHO